### PR TITLE
ACS-6300 Add a static-name flag for reportportal-prepare action

### PR DESCRIPTION
**Pull request by wojtekswieton**

### CODE REVIEW

File: .github/actions/reportportal-prepare/action.yml [hunk 1]
```diff
@@ -17,6 +17,11 @@ inputs:
   rp-extra-attributes:
     description: Extra attributes for Report Portal
     required: false
+  rp-static-name:
```
Comment (dsibilio) [inline .github/actions/reportportal-prepare/action.yml:20]:
```suggestion
  rp-static-launch-name:
```

A more descriptive name might help.

File: .github/actions/reportportal-prepare/action.yml [hunk 2]
```diff
@@ -17,6 +17,11 @@ inputs:
...
+    description: >-
+      If set, it uses RP_LAUNCH_PREFIX as RP_LAUNCH_KEY.
```
Comment (dsibilio) [inline .github/actions/reportportal-prepare/action.yml:22]:
```suggestion
    description: If set, the rp-launch-prefix will be used as the full launch name in Report Portal
```

Comment (dsibilio) [reply .github/actions/reportportal-prepare/action.yml:22]: ^ would make it easier to understand from a consumer-perspective

**Review by MichalFidor — APPROVED** [review]
lgtm

File: .github/actions/reportportal-prepare/action.yml [hunk 3]
```diff
@@ -54,4 +59,5 @@ runs:
         RP_PROJECT: ${{ inputs.rp-project }}
         RP_EXTRA_ATTRIBUTES: ${{ inputs.rp-extra-attributes }}
         AUTO: ${{ inputs.auto-configure }}
+        STATIC: ${{ inputs.rp-static-name }}
```
Comment (dsibilio) [inline .github/actions/reportportal-prepare/action.yml:62]:
```suggestion
        STATIC_LAUNCH_NAME: ${{ inputs.rp-static-launch-name }}
```

File: .github/actions/reportportal-prepare/get-rp-input.sh [hunk 4]
```diff
@@ -4,7 +4,7 @@ if [[ -n "$RP_LAUNCH_PREFIX" && -n "$RP_TOKEN" && -n "$RP_URL" && -n "$RP_PROJEC
   echo "enabled=true" >> $GITHUB_OUTPUT
 
   RP_LAUNCH_KEY="$RP_LAUNCH_PREFIX"
-  if [[ "$AUTO" == "true" ]]; then
+  if [[ "$AUTO" == "true" && "$STATIC" == "false" ]]; then
```
Comment (dsibilio) [inline .github/actions/reportportal-prepare/get-rp-input.sh:7]:
```suggestion
  if [[ "$AUTO" == "true" && "$STATIC_LAUNCH_NAME" == "false" ]]; then
```

**Review by dsibilio — CHANGES_REQUESTED** [review]

File: docs/README.md [hunk 5]
```diff
@@ -950,6 +950,9 @@ This action is usually used in combination with [reportportal-summarize](#report
 In particular, this prepares maven command line options for Report Portal integration, building the endpoint, authentication, launch key, description.
 Default context information is also added (launch attributes), unless the `auto-configure` input is set to `false`.
 
+By using the `rp-static-launch-name` flag, you can determine whether 'LAUNCH_NAME' should be static or unique for each execution. By default, it is set to `false`.
```
Comment (dsibilio) [inline docs/README.md:953]:
```suggestion
By using the `rp-static-launch-name` flag, you can determine whether the launch name in Report Portal should be static or unique for each execution. By default, it is set to `false`.
```

Make it clearer for the consumer, they shouldn't have to go through the bash script implementation to understand what a new flag does.

**Review by dsibilio — CHANGES_REQUESTED** [review]

**Comment by dsibilio** [comment]
Few test failures [here](https://github.com/Alfresco/alfresco-build-tools/actions/runs/6970049366/job/18967180738?pr=386#step:4:17) need to be looked at and addressed.

File: .github/actions/reportportal-prepare/get-rp-input.sh [hunk 6]
```diff
@@ -4,7 +4,7 @@ if [[ -n "$RP_LAUNCH_PREFIX" && -n "$RP_TOKEN" && -n "$RP_URL" && -n "$RP_PROJEC
   echo "enabled=true" >> $GITHUB_OUTPUT
 
   RP_LAUNCH_KEY="$RP_LAUNCH_PREFIX"
-  if [[ "$AUTO" == "true" ]]; then
+  if [[ "$AUTO" == "true" && "$STATIC_LAUNCH_NAME" == "false" ]]; then
```
Comment (atchertchian) [inline .github/actions/reportportal-prepare/get-rp-input.sh:7]:
```suggestion
  if [[ "$AUTO" == "true" && "$USE_STATIC_LAUNCH_NAME" == "false" ]]; then
```

**Review by dsibilio — CHANGES_REQUESTED** [review]
https://github.com/Alfresco/alfresco-build-tools/pull/386#issuecomment-1824409670

**Review by dsibilio — APPROVED** [review]
LGTM ✔️ 

_Please, next time don't delete the PR template but fill it appropriately. It's there for a reason 👍_

@atchertchian thank you very much for your help!

**Review by gionn — APPROVED** [review]
> Please, next time don't delete the PR template but fill it appropriately. It's there for a reason 👍

**Review by gionn — APPROVED** [review]

**Pull request merged**